crates/sui-core/src/consensus_adapter.rs

```diff
@@ -471,6 +466,46 @@ impl ConsensusAdapter {
         )
     }
 
+    // According to the throughput profile we want to either allow some transaction duplication or not)
+    // When throughput profile is Low and the validator is in position = 1, then it will submit to consensus with much lower latency.
+    // When throughput profile is High then we go back to default operation and no-one co-submits.
+    fn override_by_throughput_profiler(&self, position: usize, latency: Duration) -> Duration {
+        const LOW_THROUGHPUT_LATENCY_MS: u64 = 1250;
+
+        let p = self.consensus_throughput_profiler.load();
+
+        if let Some(profiler) = p.as_ref() {
+            let (level, _) = profiler.throughput_level();
+
+            // we only run this for the position = 1 validator to co-submit with the validator of
+            // position = 0. We also enable this only when the feature is enabled on the protocol config.
+            if self.protocol_config.throughput_aware_consensus_submission()
+                && position == 1
+                && level == Level::Low
+            {
+                return Duration::from_millis(LOW_THROUGHPUT_LATENCY_MS);
```

Comment:
We probably want to use `min(LOW_THROUGHPUT_LATENCY_MS, latency)`

Reply:
We can certainly do, with the current setup we don't expect `latency` to be lower than the one defined here, but we can still do.